src/gateway/delegate-routing.ts

```diff
@@ -31,11 +31,31 @@ const onlines: { [index: string]: RoutingData } = {};
 router.use(
   "/findpeer",
   mixGetPost,
+  rateLimit({
+    points: 100,
+    duration: 60*60e3,
+  }),
   asyncHandler(async (req, res, next) => {
     // @ts-ignore
-    let { id } = req.mixed;
+    const {id, timestamp, signature} = req.mixed;
 
-    if (!id) throw `Missing parameter 'id'`;
+    if (!id || !timestamp || !signature)
+      throw `Missing parameters`;
+
+    let hash = muonSha3(
+      { type: "uint64", value: timestamp },
+      { type: "string", value: "peerInfo" },
+    );
+    // @ts-ignore
+    const wallet = crypto.recover(hash, signature);
+    const requesterInfo: MuonNodeInfo[] = await NetworkIpc.filterNodes({
+      list: [wallet],
+    })!;
+    if(!requesterInfo[0])
+      throw `Invalid signature`;
+
+    if (!hasCommonContext(requesterInfo[0].id, id))
+      throw `Access denied`;
 
     const peerInfos: MuonNodeInfo[] = await NetworkIpc.filterNodes({
```

Comment:
Is it possible to call filterNodes once and get requesterInfo and peerInfos at once?
```
filterNodes({
   list: [wallet, id]
})
```